### ISSUE TITLE
Not enough examples

### ISSUE THREAD

**Issue by bernaferrari** [atlassian/pragmatic-drag-and-drop#75]
React-beautiful-dnd has many examples. Animated, input on/off, onBeforeCapture. So many things. I don't like pragmatic drag and drop because it has like 2-3 examples, they all use emotion and other things I don't use, so they are super complex, and they lack animation. So it feels a regression. From 10 LOC to 100 LOC, no animation, and harder to read code. Maybe I'm missing something, but I guess if you just had more examples, life would be easier. Or maybe you don't have examples because pragmatic is too low level and that would make making examples harder?

All I wanted is something like this:
https://react-beautiful-dnd.netlify.app/?path=/story/nested-interative-elements--stress-test

https://github.com/atlassian/pragmatic-drag-and-drop/assets/351125/265650b3-f97f-4db2-a34b-b137e4c75c74

**Comment by benjosua** [comment]
I agree, getting started with this library is incredibly difficult. I've already spent about 5 hours debugging just the examples. I hope Atlassian invests some time in improving the documentation so this library receives the recognition it deserves and doesn't end up underutilized.

**Comment by bernaferrari** [comment]
For now I'm using hello-pangea/dnd, it is not perfect, but does the job mostly well. @dnd-kit is in second place, but documentation is worse and also has bugs. I haven't found a drag and drop library that can make reorder and drop without any issues, unfortunately.

**Comment by benjosua** [comment]
After a few days of effort, I successfully implemented pdnd, and it's now running smoothly. I believe they've also expanded the documentation. Don't give up; you'll likely experience a moment of clarity as you work through it. And I tried dnd-kit before, but pdnd is just much smoother and way more performant.

**Comment by benjosua** [comment]
@bernaferrari If you need help creating the move around animations, I implemented them using framer/motion and could provide an example if you are interested.

**Comment by bernaferrari** [comment]
I still think the api is necessarily complex compared to beautiful dnd or dnd kit, but I would love this example.

**Comment by alexreardon** [comment]
Here are some simpler examples that might help you get started: https://atlassian.design/components/pragmatic-drag-and-drop/examples/#simple-list-on-other-stacks

A challenge with creating "simple" examples is that pdnd is a low level library, and it gives users a lot of control on what they want to do. It's operating at a similar abstraction level as `jQuery` (low level, smooths over browser roughness, un-opionionated about what you want to do with it)

**Comment by bernaferrari** [comment]
pdnd shouldn't replace beautiful dnd if it is extremely low level, hard to use and severely limited. They should at least have maintained or reimplemented it using pdnd.

**Comment by SimonFrank14** [comment]
> @bernaferrari If you need help creating the move around animations, I implemented them using framer/motion and could provide an example if you are interested.

Hey,
Do you mind sharing an example for this? I'm currently working with DnDkit but I am experiencing a lot of issues because my setup is a little bit complex and not so straightforward with DnDkit anymore.

This results in a lot of renders and I thought of using pDnD but the animations are a must have feature for me.

Thank you.

**Comment by bernaferrari** [comment]
@SimonFrank14 use Framer Motion. Trust me, it is the best. I tried like 5 drag and drop libraries. Nothing even comes close on simplicity + easy of use + power + what you expect.

**Comment by kvanapamala** [comment]
If anyone looking to use lists drag and drop, there is very good example laid out by Rishi Purwar. Please find 
example : https://blog.logrocket.com/implement-pragmatic-drag-drop-library-guide/
started code is in GIT: https://github.com/rishipurwar1/pragmatic-drag-and-drop-demo

**Comment by chihiroxxc** [comment]
Hi @benjosua , I would appreciate if you can share your demo with moving animation, I didn't find any example of pragmatic-drag-and-drop with moving animation.

Thanks!

**Comment by bernaferrari** [comment]
@chihiroxxc go with Framer Motion. Pragmatic was born dead.

**Comment by Shadysoftdev** [comment]
I have some progress on my Drag and drop with pragmatic its at least partially functional, but its probably very specific to my use case,

**Comment by andrew-y0u** [comment]
It's unfair to have comments like "born dead", "super complex" and "hard to read code", especially from someone who positions themselves as a software engineer. 

I’d want to say thanks for your work @alexreardon! I'm migrating from «react-beautiful-dnd» manually, without using migration components. So far so good! API is quite simple and it lets us build different custom things which is great!

**Comment by GiancarlosIO** [comment]
The library and low-level api are awesome!
But I agree it would be useful to at least have a little guidance about implementing animations. I think that's one of the missing pieces in the documentation 🙏🏼

**Comment by digrejzo** [comment]
^ agreed, better and more docs is never a bad thing, especially if it's well written ones!

On the whole "born dead" nonsense and "I need something that I can just paste into my codebase without knowing what it does", it would appear as if this simply isn't the correct tool for that kind of mindset.

I'm working on a page with multiple droppable containers - each having a grid layout of their own draggable (and droppable) cells, where these droppable parent containers also need to be reorderable, so after unsuccessfully attempting to implement dnd-kit, then also giving hello-pangea/dnd a go, and having to fight with their opinionated approach to how the app layout should look like, how the animations will behave, poorly described advanced usecases, not being able to create custom placeholders (I'm looking at you hello-pangea/dnd), how I needed to juggle several layers of Contexts around and enable/disable them on the fly, and other various things that gave me nice headaches, made my DND behavior essentially an impossible task. 

Pragmatic on the other hand, being as unopinionated as it is, allowed me to refactor the whole thing in ~2-3 days and support all the interactive DND usecases perfectly. Sure, there's a bit more work to do and there's room for better UX with some animations which you can just do on your own if you want to completely control the whole experience, but it's still very much plug-n-play and works reasonably well, especially after you spend some time with the APIs it exposes. So another honest thanks from me and keep up the good work @alexreardon ! (and sorry for the wall of text)

**Comment by darbaoui** [comment]
How can I find the timeline example that appears in the introduction GIF at the end?

**Comment by warm-tune** [comment]
> For now I'm using hello-pangea/dnd, it is not perfect, but does the job mostly well. [@dnd-kit](https://github.com/dnd-kit) is in second place, but documentation is worse and also has bugs. I haven't found a drag and drop library that can make reorder and drop without any issues, unfortunately.

dnd-kit is abandoned. do not start new projects with it, you will never get any issues fixed.

**Comment by jikyu** [comment]
> > For now I'm using hello-pangea/dnd, it is not perfect, but does the job mostly well. [@dnd-kit](https://github.com/dnd-kit) is in second place, but documentation is worse and also has bugs. I haven't found a drag and drop library that can make reorder and drop without any issues, unfortunately.
> 
> dnd-kit is abandoned. do not start new projects with it, you will never get any issues fixed.

@warm-tune please stop spreading misinformation and hateful dialogue.

As noted [here](https://github.com/clauderic/dnd-kit/issues/1194#issuecomment-1696704815) the library's work is in active development in the [experimental branch](https://github.com/clauderic/dnd-kit/tree/experimental). Library maintainer is super active in discussion and have been fixing bugs.
